src/goose/brokers/redis/commands.clj

```diff
@@ -251,11 +251,33 @@
   (wcar* conn (car/zremrangebyscore sorted-set sorted-set-min score)))
 
 ;;; ============ Hashes ============
-
 (defn parse-map [conn hash]
   (wcar* conn (car/parse-map (car/hgetall hash) :keywordize)))
 
 ;;; ============ Misc ============
-
 (defn exists [conn key]
   (wcar* conn (car/exists key)))
+
+;;; ============ Batch ============
+(defn get-batch-state
+  [conn id]
+  (let [batch-state-key (d/prefix-batch id)
+        enqueued-job-set (d/construct-batch-job-set id d/enqueued-job-set)
+        retrying-job-set (d/construct-batch-job-set id d/retrying-job-set)
+        successful-job-set (d/construct-batch-job-set id d/successful-job-set)
+        dead-job-set (d/construct-batch-job-set id d/dead-job-set)
+        [_ [batch-state enqueued retrying successful dead]]
+        (car/atomic conn
```

Comment:
I understand the intent behind using a transaction here is to get multiple queries as part of 1 command instead of calling it multiple times?
Would an EVAL script be preferable to using a transaction here?

Reply:
Can defer this to Neena/Sezal. I don't know what's the **correct** way to do this in Redis.

Reply:
Discussed about using EVAL here with @sezaljain. And we concluded that it might be overkill for these set of commands.